### DISCUSSION TITLE
Any document or guide on the credential issuance transportation

### DISCUSSION THREAD

**Discussion by thomas-tran** [decentralized-identity/veramo#650] [Documentation]
Assume I had setup the issuer agent on the cloud and would like to issue the credential to the holder which has the agent on the mobile device. How can the holder received the credential? What's the transportation e.g. https, web socket will be used? If the holder is not available at the time the credential was issued. How to handle the credential message?

**Comment by mirceanis** [comment]
Hi @thomas-tran,

For the moment we can't provide too much guidance on how to handle the offline agent situation.
In these early versions we have left a lot of the transportation burden on the application that uses veramo to run an agent, in which case veramo is unaware of exactly how the message got to it as long as the application calls `agent.handleMessage()`.

We provide the `@veramo/remote-server` package that has some `express.js` routers available that can act as HTTP ingress endpoints for messaging, and we are also working on other modes of transport for DIDComm.
This doesn't solve the offline agent problem, though; and I suspect it will not be resolved purely at the framework level, but more of a protocol level.

We could speculate that in many situations the issuer can keep the credential until the holder requests it, but of course this cannot be generalized.
Another approach is to use "inbox agents" whose only role would be to stay online to act as service endpoints until the "real holder agent" comes online and downloads their messages for processing.
